### PR TITLE
Update Readme.md

### PR DESCRIPTION
**Pull request by SvetlanaMikheeva**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
 <!-- default badges end -->
-
-<!-- default file list -->
-*Files to look at*:
-
-* [Default.aspx](./CS/Default.aspx) (VB: [Default.aspx](./VB/Default.aspx))
-* [Default.aspx.cs](./CS/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/Default.aspx.vb))
-<!-- default file list end -->
-# How to use ASPxRichEdit to edit RTF data in ASPxGridView's EditForm
+# Grid View for ASP.NET Web Forms - How to use the Rich Text Editor to edit RTF data in the Edit Form
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: Just to satisfy my curiosity -- I've never encountered this curious term -- RTF data. Do you use it everywhere throughout your documentation repository?

Why do I ask? What is RTF data? Basically, RTF data is formatted text. Did you say "RTF data" because you wanted to emphasize the fact that formatted text is stored in a database? If this fact is really that important in this context, perhaps, there is a better way to emphasize it.

Either way, in my personal opinion, the phrase "RTF data" sounds strange. I'd like to hear your opinion on this.

File: Readme.md [hunk 2]
```diff
@@ -1,30 +1,51 @@
...
 <!-- run online -->
 **[[Run Online]](https://codecentral.devexpress.com/t260978/)**
 <!-- run online end -->
 
+This example demonstrates how to add the [Rich Text Editor](https://docs.devexpress.com/AspNet/17721/components/rich-text-editor) to the [Grid View](https://docs.devexpress.com/AspNet/5823/components/grid-view) control's edit form. The Rich Text Editor displays RTF data from the data source bound to the Grid View and allows users to modify this data.
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: I read your text further... Again, I find this strongly unnatural when you refer to formatted text, Rich text as "RTF data". When users "modify RTF data", they just edit text -- no more, no less. This is a very complicated way to say such a simple thing. Besides, the very name of your Rich Text Editor already defines sufficient context. Then, you explicitly say that the component displays your formatted text from the data source bound to the Grid View.

```suggestion
This example demonstrates how to add the [Rich Text Editor](https://docs.devexpress.com/AspNet/17721/components/rich-text-editor) to the [Grid View](https://docs.devexpress.com/AspNet/5823/components/grid-view) control's edit form. The Rich Text Editor displays formatted text loaded from the data source bound to the Grid View and allows users to edit this text.
```

File: Readme.md [hunk 3]
```diff
@@ -1,30 +1,51 @@
...
+
+> **Note**  
+> Data modification is not allowed in an online version of this example. To allow editing, download the example and comment out the [throw](./CS/Default.aspx.cs#L97) expression in the `RowUpdating` event handler.
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: > Data modification is not allowed in **an** online version of this example.

No, you are referring to a quite specific version of this known example. What version of this example? The online version of this example. Even if we suppose that we can choose between different versions (23.1, 22.2, etc.), we have only one "current" version. The important thing that this version exists online.

```suggestion
> Data modification is not allowed in the online version of this example. To allow edit operations, download the example and comment out the [throw](./CS/Default.aspx.cs#L97) expression in the `RowUpdating` event handler.
```

File: Readme.md [hunk 4]
```diff
@@ -1,30 +1,51 @@
...
+2. Add the [Rich Text Editor](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit) control to the [EditItemTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.EditItemTemplate?p=netframework) of the column that contains RTF data.
+
+3. Handle the Rich Text Editor control's [Init](https://learn.microsoft.com/en-us/dotnet/api/system.web.ui.control.init?view=netframework-4.8.1) event. If the event occurs when a new row is being edited, assign a new unique identifier to the Rich Text Editor's [DocumentId](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.DocumentId?p=netframework) property. If an existing row is being edited, get RTF data from the corresponding cell and pass the data and the row key to the control's [Open](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.Open(System.String-DevExpress.XtraRichEdit.DocumentFormat-System.Func-System.Byte---)) method.
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: From this paragraph, I cannot understand whether this event occurs when a user activates an in-place cell editor/row editor or the event occurs when a row is being edited for some time. Right now, your wording suggests the latter. More clarity here is in order. In my opinion, this event occurs when a user activates an in-place editor, and the editor itself is about to become active. That is, the editing process is about to start at the moment of this event occurrence.

File: Readme.md [hunk 5]
```diff
@@ -1,30 +1,51 @@
...
+
+4. Handle the Grid View's [RowInserting](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.RowInserting?p=netframework) and [RowUpdating](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.RowUpdating?p=netframework) events. In both event handlers, call the Rich Text Editor's [SaveCopy](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.SaveCopy(DevExpress.XtraRichEdit.DocumentFormat)) method to get document content in RTF, then save the content back to the data source.
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: > to get document content in RTF

I do not understand this phrase at all, therefore, I cannot suggest an edit.

File: Readme.md [hunk 6]
```diff
@@ -1,30 +1,51 @@
...
+
+5. To prevent synchronization problems, handle the Rich Text Editor's client-side [Init](https://docs.devexpress.com/AspNet/js-ASPxClientControlBase.Init) event and the Grid View's client-side [BeginCallback](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.BeginCallback) event as follows:
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: As Ray says, we are not in problem-solving business. A "Problem" is a real-life problem. An "Issue" is a "technical problem".

We never say "problem". We may say "problematic behavior", however.

```suggestion
5. To prevent synchronization issues, handle the Rich Text Editor's client-side [Init](https://docs.devexpress.com/AspNet/js-ASPxClientControlBase.Init) event and the Grid View's client-side [BeginCallback](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.BeginCallback) event as follows:
```

**Review by dmitry-eliseev-devexpress — COMMENTED** [review]
**Structure and Semantics**

This is a simple step-by-step instruction topic with a code example. I found no structure-related issues. Well done. However, a couple of paragraphs lacks clarity and unambiguity. I also noticed a potential technical correctness issue (see the corresponding comment). You need to either clarify or rewrite them.

One more thing -- this strange term "RTF data". When you describe an entirely technical aspect -- RTF data sounds fine ("the column that contains RTF data", for example). However, when you describe thing from the user perspective, "RTF data" sounds strange. A user does not "modify RTF data in a column", a user edits formatted text. That's what important here.

As I said before, I never encountered this term "RTF data" in our documentation. This fact raised a flag for me.

**Language**

In general, your language is good. I only made one edit regarding an article and one more edit regarding word choice. As I said, we do not use the word "problem" because in English it has subtext of a real-life problem rather than a technical issue. The word "issue" has no such subtext, that's why it's recommended to use it.

**Bottom Line**

I cannot approve this topic until we clarify critical issues.

File: Readme.md (hunk 1 above)
Comment (SvetlanaMikheeva) [reply Readme.md]: Yes, I mean formatted text that is stored in a database. I like your suggestion to use the "formatted text" phrase instead the "RTF data". If it's okay, I'll replace the "RTF data" in the entire Readme text.

And no, we do not use RTF data throughout the documentation.

File: Readme.md (hunk 4 above)
Comment (SvetlanaMikheeva) [reply Readme.md:23]: When a user clicks the **New** or **Edit** button in the command column, the edit form appears. This form contains the Rich Text Editor, so after the edit form appears, the Rich Text Editor's `Init` event occurs. To check whether a user clicked the **New** or **Edit** button, the example uses the Grid View's IsNewRowEditing property, what's why I say "is being edited". However, the row is rather about to being edited.

```suggestion
3. Handle the Rich Text Editor control's [Init](https://learn.microsoft.com/en-us/dotnet/api/system.web.ui.control.init?view=netframework-4.8.1) event that occurs after the edit form appears. Use the Grid View's [IsNewRowEditing](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.IsNewRowEditing?p=netframework) to determine whether the edit form after a user clicked the **New** or **Edit** button. If the user cliked **New** button, assign a new unique identifier to the Rich Text Editor's [DocumentId](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.DocumentId?p=netframework) property. Otherwise, get RTF data from the corresponding cell and pass the data and the row key to the control's [Open](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.Open(System.String-DevExpress.XtraRichEdit.DocumentFormat-System.Func-System.Byte---)) method. 
```

Comment (SvetlanaMikheeva) [reply Readme.md]: When a user clicks the **New** or **Edit** button in the command column, the edit form appears. This form contains the Rich Text Editor, so after the edit form appears, the Rich Text Editor's `Init` event occurs. To check whether a user clicked the **New** or **Edit** button, the example uses the Grid View's IsNewRowEditing property, what's why I say "is being edited". However, the row is rather about to being edited.

```suggestion
3. Handle the Rich Text Editor control's [Init](https://learn.microsoft.com/en-us/dotnet/api/system.web.ui.control.init?view=netframework-4.8.1) event that occurs after the edit form appears. Use the Grid View's [IsNewRowEditing](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.IsNewRowEditing?p=netframework) property to determine whether the edit form appeared after a user clicked the **New** or **Edit** button. If the user cliked the **New** button, assign a new unique identifier to the Rich Text Editor's [DocumentId](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.DocumentId?p=netframework) property. Otherwise, get formatted text string from the corresponding cell and pass the string and the row key to the control's [Open](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.Open(System.String-DevExpress.XtraRichEdit.DocumentFormat-System.Func-System.Byte---)) method. 
```

File: Readme.md (hunk 5 above)
Comment (SvetlanaMikheeva) [reply Readme.md]:
```suggestion
4. Handle the Grid View's [RowInserting](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.RowInserting?p=netframework) and [RowUpdating](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.RowUpdating?p=netframework) events. In both event handlers, call the Rich Text Editor control's [SaveCopy](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.SaveCopy(DevExpress.XtraRichEdit.DocumentFormat)) method to get content of the document opened in the control as an array of bytes. Save the array back to the data source.
```

or maybe 

```suggestion
4. Handle the Grid View's [RowInserting](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.RowInserting?p=netframework) and [RowUpdating](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.RowUpdating?p=netframework) events. In both event handlers, call the Rich Text Editor's [SaveCopy](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.SaveCopy(DevExpress.XtraRichEdit.DocumentFormat)) method to get formatted text displayed in the Rich Text Editor. Save the text back to the data source.
```

File: Readme.md (hunk 1 above)
Comment (dmitry-eliseev-devexpress) [reply Readme.md]: Good call. Thank you. This "RTF data" is basically an RTF string stored in a BLOB database field.

File: Readme.md (hunk 4 above)
Comment (dmitry-eliseev-devexpress) [reply Readme.md]: Yes. Your explanations make sense indeed. Thank you. However, I suggest a number of adjustments for your latest version:

> cliked

A typo.

> determine

When you say "determine" -- it more says of modification -- you determine its state, appearance, that is, actively affect it. A synonym for "determine" is "specify", for example. However, in this case, you just need to know what it is. Therefore, you "identify", that is, learn something.

> Otherwise, get formatted text string from the corresponding cell

Since "string" is countable, you've missed an article.

```suggestion
3. Handle the Rich Text Editor control's [Init](https://learn.microsoft.com/en-us/dotnet/api/system.web.ui.control.init?view=netframework-4.8.1) event that occurs after the edit form appears. Use the Grid View's [IsNewRowEditing](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.IsNewRowEditing?p=netframework) property to identify if the edit form appeared in response to a click on the **New** or **Edit** button. If a user clicked the **New** button, assign a new unique identifier to the Rich Text Editor's [DocumentId](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.DocumentId?p=netframework) property. Otherwise, get a formatted text string from the corresponding cell and pass the string and the row key to the control's [Open](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.Open(System.String-DevExpress.XtraRichEdit.DocumentFormat-System.Func-System.Byte---)) method.
```

File: Readme.md (hunk 5 above)
Comment (dmitry-eliseev-devexpress) [reply Readme.md]: I like your first version for its technical details.

**Review by dmitry-eliseev-devexpress — APPROVED** [review]
